### PR TITLE
OCT-16: fix Marketplace page error 400

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**
image
image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Comment by tseho** [comment]
What's the behavior if the marketplace is down with your new version ?

**Comment by guillaumefer-akeneo** [comment]
> What's the behavior if the marketplace is down with your new version ?

image

The UnreachableMarketplace component is no longer rendered for the moment.

**Comment by tseho** [comment]
That's not good :D

When we catch the exception
```
} catch (\Exception $e) {
    $this->logger->error(\sprintf('unable to retrieve the list of apps, got error "%s"', $e->getMessage()));

    return new Response(null, Response::HTTP_NO_CONTENT);
}
```

Is there a way to catch a different exception for 400 errors or check the given exception if it's a 400 error and only then, return an empty list but keep the default behavior of sending `new Response(null, Response::HTTP_NO_CONTENT);` for all other errors ?

**Comment by guillaumefer-akeneo** [comment]
> That's not good :D
> 
> When we catch the exception
> 
> ```
> } catch (\Exception $e) {
>     $this->logger->error(\sprintf('unable to retrieve the list of apps, got error "%s"', $e->getMessage()));
> 
>     return new Response(null, Response::HTTP_NO_CONTENT);
> }
> ```
> 
> Is there a way to catch a different exception for 400 errors or check the given exception if it's a 400 error and only then, return an empty list but keep the default behavior of sending `new Response(null, Response::HTTP_NO_CONTENT);` for all other errors ?

It's fixed :D

**Comment by guillaumefer-akeneo** [comment]
> That's not good :D
> 
> When we catch the exception
> 
> ```
> } catch (\Exception $e) {
>     $this->logger->error(\sprintf('unable to retrieve the list of apps, got error "%s"', $e->getMessage()));
> 
>     return new Response(null, Response::HTTP_NO_CONTENT);
> }
> ```
> 
> Is there a way to catch a different exception for 400 errors or check the given exception if it's a 400 error and only then, return an empty list but keep the default behavior of sending `new Response(null, Response::HTTP_NO_CONTENT);` for all other errors ?

it's fixed

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/Controller/Internal/GetAllAppsAction.php [hunk 1]
```diff
@@ -47,6 +47,10 @@ public function __invoke(Request $request): Response
         } catch (\Exception $e) {
             $this->logger->error(\sprintf('unable to retrieve the list of apps, got error "%s"', $e->getMessage()));
 
+            if (Response::HTTP_BAD_REQUEST === $e->getCode()) {
+                return new JsonResponse(GetAllAppsResult::create(0, [])->normalize());
+            }
+
```
Comment (masacc) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/Controller/Internal/GetAllAppsAction.php:53]: Are there some tests to create/update for this use case?

**Review by willy-ahva — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

**Review by masacc — APPROVED** [review]

**Review by masacc — APPROVED** [review]

**Pull request merged**